src/core/public/chrome/ui/header/header_search_bar.tsx

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiButtonIcon,
+  EuiFieldSearch,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiListGroup,
+  EuiListGroupItem,
+  EuiPanel,
+  EuiPopover,
+  EuiText,
+  EuiTitle,
+  EuiToolTip,
+} from '@elastic/eui';
+import React, { ReactNode, useRef, useState } from 'react';
+import { i18n } from '@osd/i18n';
+import { GlobalSearchCommand, SearchObjectTypes } from '../../global_search';
+
+interface Props {
+  globalSearchCommands: GlobalSearchCommand[];
+  panel?: boolean;
+  onSearchResultClick?: () => void;
+}
+
+/**
+ * search input match with `@` will handled by saved objects search command
+ * search input match with `>` will handled by plugin customized commands
+ */
+export const SAVED_OBJECTS_SYMBOL = '@';
+export const COMMANDS_SYMBOL = '>';
+
+export const SearchCommandFilters = {
+  [SearchObjectTypes.PAGES]: (value: string) => {
+    return {
+      match: !value.startsWith(SAVED_OBJECTS_SYMBOL) && !value.startsWith(COMMANDS_SYMBOL),
+      searchValue: value,
+    };
+  },
+  [SearchObjectTypes.SAVED_OBJECTS]: (value: string) => {
+    return {
+      match: value.startsWith(SAVED_OBJECTS_SYMBOL),
+      searchValue: value.replace(SAVED_OBJECTS_SYMBOL, '').trim(),
+    };
+  },
+};
```

Comment:
That's make sense, to let each command decide what's the behavior. But for builtin pages and saved objects search command, we don't want to provide the flexibility to customized the alias, name and behavior of whether to run the command or not.

Reply:
>If the input not start with any command name/alias, the it should call run for all commands with the current input. 

I think we should search pages only, to search other types it will need explicit type the alias